### PR TITLE
Fix UpdateBalancingState not called when address attributes are modified

### PR DESCRIPTION
**Pull request by JamesNK**
gRPC load balancing would update subchannels with modified address attributes when they changed, but the overall channel state was never updated.

### CODE REVIEW

**Comment by JamesNK** [comment]
Ready for review. I know grpc-web test is failing. Will fix it once RC2 is public.

File: global.json [hunk 1]
```diff
@@ -1,6 +1,5 @@
 {
   "sdk": {
-    "version": "9.0.100-preview.7.24407.12",
```
Comment (amcasey) [inline global.json:3]: What's the significance of removing `rollForward`?

File: src/Grpc.Net.Client/Balancer/SubchannelsLoadBalancer.cs [hunk 2]
```diff
@@ -122,6 +122,7 @@ public override void UpdateChannelState(ChannelState state)
 
         var allUpdatedSubchannels = new List<AddressSubchannel>();
         var newSubchannels = new List<Subchannel>();
+        var modifiedSubchannelsCount = 0;
```
Comment (amcasey) [inline src/Grpc.Net.Client/Balancer/SubchannelsLoadBalancer.cs:125]: It doesn't seem important either way, but it looks like this could be a simple bool.

**Review by amcasey — APPROVED** [review]
The code seems to match the description and the description seems reasonable.

File: global.json (hunk 1 above)
Comment (JamesNK) [reply global.json:3]: RC2 preview was causing trouble on my machine.

**Pull request merged**